### ISSUE TITLE
[Bug Report] BaseDevice / DeviceParser silently change connectionStateUpdatedTime from UTC to a Local Time string

### ISSUE THREAD

**Issue by derSchtefan** [Azure/azure-iot-sdk-java#1414]
# Context

- **Java runtime used:** 11
- **SDK version used:** iot-service-client:1.33.2

## Description of the issue
The constructor of BaseDevice in conjunction with DeviceParser causes the connectionStateUpdatedTime field to end up as a local time based string, without reason or indication. This is a problem because:
* All other fields in the SDK offer UTC 
* The way back to a UTC timestamp is awkward and fails if executed during DST change
* This field is used to order events and conditionally update fields in a database, local/UTC timestamp mismatches caused a bug in our solution.

Proposed solution: let the faulty String stay as it is (too much code might rely on that quirky behaviour), but expose the Date object from DeviceParser on BaseDevice.

## Code sample exhibiting the issue

At first `DeviceParser.java` behaves correctly
```java
191:  if (deviceParser.connectionStateUpdatedTimeString != null) {
192:      this.connectionStateUpdatedTimeString = deviceParser.connectionStateUpdatedTimeString;
193:      this.connectionStateUpdatedTime = ParserUtility.getDateTimeUtc(deviceParser.connectionStateUpdatedTimeString);
194:  }
```

With an example of the result being:
```
String connectionStateUpdatedTimeString = 2021-11-17T05:23:04.3511935Z
Date   connectionStateUpdatedTime       = Wed Nov 17 06:23:04 CET 2021  
```

The issue happens then in `BaseDevice.java`:
```java
459:  if (parser.getConnectionStateUpdatedTime() != null)
460:  {
461:      this.connectionStateUpdatedTime = ParserUtility.getDateStringFromDate(parser.getConnectionStateUpdatedTime());
462:  }
```

Resulting in 
```
String connectionStateUpdatedTime       = 2021-11-17T06:23:04
```

Notice how the date suddenly transformed from 5am to 6am, with no reason why it also suddenly has the fractional part truncated.

**Comment by timtay-microsoft** [comment]
Yep, this looks like a bug. I'll try to get a fix checked in soon.

**Comment by timtay-microsoft** [comment]
After some discussion internally, we're worried about this fix being a breaking change for customers. We're planning on bringing it to our upcoming 2.X.X versions of the clients though since we agree the SDK should not be doing this.

**Comment by derSchtefan** [comment]
Yes, but it will also be a breaking change for your 2.0 SDK, since nobody will change their code.

That's why I suggested to add a second field of type Instant (or any other type) with a proper name. DeviceParser also has two fields, so it's anyway weird why this second field was dropped.

**Comment by timtay-microsoft** [comment]
> Yes, but it will also be a breaking change for your 2.0 SDK, since nobody will change their code.

Users upgrading to the 2.0 SDK will be have to change their code as there are a number of other breaking changes involved in it regardless of where we put this fix.

**Comment by timtay-microsoft** [comment]
We're planning on releasing the v2 clients within this week, and they will contain this fix

**Comment by timtay-microsoft** [comment]
The newly released Device Client version 2.0.0 has the fix for this issue, so I'm closing this thread.
